### PR TITLE
Added option to toggle if linting errors should break the build

### PR DESCRIPTION
**Pull request by boedlen**

### CODE REVIEW

File: src/TSLint.MSBuild.targets [hunk 1]
```diff
@@ -34,9 +35,14 @@
     <!-- Run TSLint via the runner -->
     <Exec 
       Command="&quot;$(TSLintNodeExe)&quot; --harmony &quot;$(TSLintRunnerScript)&quot; -exclude &quot;$(TSLintExclude)&quot; -files-root-dir &quot;$(TSLintFilesRootDir)&quot; -file-list-file &quot;$(TSLintFileListFile)&quot;"
-      IgnoreExitCode="true" />
+      IgnoreExitCode="true">
+      <Output TaskParameter="ExitCode" PropertyName="ErrorCode"/>
```
Comment (JoshuaKGoldberg) [inline src/TSLint.MSBuild.targets]: Nit: space before `/>`

File: src/TSLint.MSBuild.targets [hunk 2]
```diff
@@ -34,9 +35,14 @@
...
+    </Exec>
 
     <!-- Clean up our compiled file list when done -->
     <Delete Condition="'$(TSLintDeleteFileListFile)' == 'true'" Files="$(TSLintFileListFile)" />
+
+    <!-- Return an Error if linter returned exitcode -1 and we should break on errors -->
+    <Error Condition="'$(ErrorCode)' == '-1' And '$(TSLintBreakBuildOnError)' == 'true'" />
```
Comment (JoshuaKGoldberg) [inline src/TSLint.MSBuild.targets]: Nit: does the `And` need to have a capital A?

File: src/TSLint.MSBuild.targets [hunk 3]
```diff
@@ -34,9 +35,14 @@
...
       Command="&quot;$(TSLintNodeExe)&quot; --harmony &quot;$(TSLintRunnerScript)&quot; -exclude &quot;$(TSLintExclude)&quot; -files-root-dir &quot;$(TSLintFilesRootDir)&quot; -file-list-file &quot;$(TSLintFileListFile)&quot;"
-      IgnoreExitCode="true" />
+      IgnoreExitCode="true">
+      <Output TaskParameter="ExitCode" PropertyName="ErrorCode"/>
+    </Exec>
 
     <!-- Clean up our compiled file list when done -->
     <Delete Condition="'$(TSLintDeleteFileListFile)' == 'true'" Files="$(TSLintFileListFile)" />
+
+    <!-- Return an Error if linter returned exitcode -1 and we should break on errors -->
```
Comment (JoshuaKGoldberg) [inline src/TSLint.MSBuild.targets]: Nit: lower-case E in `Error`

**Comment by JoshuaKGoldberg** [comment]
A few small nits, otherwise looks great!

**Comment by JoshuaKGoldberg** [comment]
Fixes #29.

**Comment by boedlen** [comment]
Fixed the nits

**Comment by JoshuaKGoldberg** [comment]
Tested it out locally, looks great! Thanks so much for contributing :)

**Pull request merged**